subsys/usb/device/class/msc.c

```diff
@@ -373,22 +393,24 @@ static bool readCapacity(void)
 
 static void thread_memory_read_done(void)
 {
-	uint32_t n;
+	uint32_t n = length;
 
-	n = (length > MAX_PACKET) ? MAX_PACKET : length;
-	if ((addr + n) > memory_size) {
-		n = memory_size - addr;
-		stage = MSC_ERROR;
-	}
+	if (n > MAX_PACKET)
+		n = MAX_PACKET;
+	if (n > BLOCK_SIZE - curr_offset)
+		n = BLOCK_SIZE - curr_offset;
```

Comment:
```c
	if (n > MAX_PACKET) {
		n = MAX_PACKET;
	}

	if (n > BLOCK_SIZE - curr_offset) {
		n = BLOCK_SIZE - curr_offset;
	}
```

Reply:
@taralx our compliance checks don't verify this, but you should use braces in your `if` statements as @jfischer-no pointed out above